### ISSUE TITLE
URL.openStream() may cause a long timeout

### ISSUE THREAD

**Issue by StefanFranzWeiser** [aposin/LicenseScout#12]
## Bug report
`org.aposin.licensescout.license.LicenseUtil.addLicensesFromPom(...)` uses `URL.openStream()` to download pom files from the network. If URLs are not available (e.g. pom file for artifact not existing) a default timeout of 20 sec is used if no system properties `sun.net.client.defaultReadTimeout` and `sun.net.client.defaultConnectTimeout` are used. The default behaviour of 20 sec timeout is not practical as setting the system property may effect other maven plugins with unexpected behaviour.

## Expected Behavior - What where you expecting to happen?

When pom files are not available a timeout of 1-2 sec should be expected (instead of 20 sec each). 

## Current Behavior - What happens?

LicenseScout takes much time for execution when poms cannot be found. 
Pom files are small artifacts and do not need much time for download so a timeout of 20 sec is far too high.

## Possible Solution

Instead of 
```
URL url = new URL(urlString);
url.openStream();
```

use 
```
URL url = new URL(urlString);
URLConnection connection = url.openConnection();
connection.setConnectTimeout(1000);
connection.getInputStream();
```

## What are the steps to reproduce this issue? (for bugs)

Execute `new URL("https://repo.maven.apache.org/maven2/org/eclipse/orbit/bundles/google/1.0.0-SNAPSHOT/google-1.0.0-SNAPSHOT.pom").openStream()`;

## Context

Querying an Eclipse product for licenses containing com.google.guava. 

## Your Environment

* Version used: 1.3.0
* Environment name and version (e.g. Chrome 39, node.js 5.4): jdk-11.0.3+7 (AdoptOpenJDK)
* Operating System and version (desktop or mobile): Win10

**Comment by pfistererm** [comment]
A solution should include a maven parameter that allows to specify the duration of the timeout - with a reasonable default value.

**Comment by pfistererm** [comment]
An alternative solution (specific for this case) may be to check if Maven central (or the configured proxy repository) is reachable under the configured base URL. If not, a warning is written to the log and POM lookup is disabled for this LicenseScout run. That may be more clear to the user in a sense of giving feedback what is going on and what the root cause of a problem may be. This is especially desirable since the license report may change significantly if access to Maven central was not possible (License information not detectable for certain artifacts due to missing information), and this change is not obvious from the license report. I actually consider this issue so significant that in my opinion the license report should include a remark if POM lookup was working.

**Comment by pfistererm** [comment]
Closed since #28 implements both of the suggested solutions.
